code/core/system/proxy-velodyne64/testsuites/ProxyVelodyne64TestSuite.h

```diff
@@ -22,21 +22,202 @@
+#include "opendavinci/odcore/io/conference/ContainerConference.h"
+#include "opendavinci/odcore/io/conference/ContainerListener.h"
+#include "opendavinci/odcore/io/protocol/PCAPProtocol.h"
+#include "opendavinci/odcore/wrapper/CompressionFactory.h"
+#include "opendavinci/odcore/wrapper/DecompressedData.h"
+#include "opendavinci/odcore/wrapper/SharedMemory.h"
+#include "opendavinci/odcore/wrapper/SharedMemoryFactory.h"
+
+#include "../include/velodyne64Decoder.h"
 
 using namespace std;
-using namespace opendlv::core::system::proxy;
+using namespace odcore::base;
+using namespace odcore::data;
+using namespace odcore::io::protocol;
+using namespace odcore::wrapper;
+
+class MyContainerConference : public odcore::io::conference::ContainerConference {
+   public:
+    MyContainerConference(std::shared_ptr< odcore::wrapper::SharedMemory > m)
+        : ContainerConference()
+        , m_counter(0)
+        , m_velodyneMemory(m) {}
+
+    virtual void send(odcore::data::Container &c) const {
+        if (c.getDataType() == odcore::data::SharedPointCloud::ID()) {
+        }
+        m_counter++;
+        if (m_counter == 2) { //We are only interested in the x, y, z, intensity values of points in the second Velodyne frame (Frame 1)
+            if (c.getDataType() == odcore::data::SharedPointCloud::ID()) {
+                odcore::data::SharedPointCloud velodyneFrame = c.getData< SharedPointCloud >();                                            //Get shared point cloud
+                std::shared_ptr< odcore::wrapper::SharedMemory > vsm = SharedMemoryFactory::attachToSharedMemory(velodyneFrame.getName()); //Attach the shared point cloud to the shared memory
+
+                if (vsm.get() != NULL) {
+                    if (vsm->isValid()) {
+                        //odcore::base::Lock l(vsm); //No need to lock shared resource as the test suite runs in a single thread. If the lock is enabled, the test suite will not terminate
+                        if (velodyneFrame.getComponentDataType() == SharedPointCloud::FLOAT_T && (velodyneFrame.getNumberOfComponentsPerPoint() == 4) && (velodyneFrame.getUserInfo() == SharedPointCloud::XYZ_INTENSITY)) {
+                            memcpy(m_velodyneMemory->getSharedMemory(), vsm->getSharedMemory(), velodyneFrame.getSize());
+                            cout << "Copy decoded data in Frame 1 to a memory segment" << endl;
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    mutable uint8_t m_counter;
+    mutable std::shared_ptr< odcore::wrapper::SharedMemory > m_velodyneMemory;
+};
+
+class packetToByte : public odcore::io::conference::ContainerListener {
+   public:
+    packetToByte(std::shared_ptr< odcore::wrapper::SharedMemory > m1, std::shared_ptr< odcore::wrapper::SharedMemory > m2)
+        : mcc(m2)
+        , v64decoder(m1, mcc, "../db.xml") //The calibration file db.xml is automatically copied to the parent folder of the test suite binary
```

Comment:
Good! With the "../db.xml" parameter, the test case, source code, and CMakeLists.txt are clean.